SDWebImage/SDWebImageDownloaderOperation.m

```diff
@@ -65,6 +65,11 @@ @implementation SDWebImageDownloaderOperation
 @synthesize executing = _executing;
 @synthesize finished = _finished;
 
+- (void)dealloc {
+    // Edge case if user call [SDWebImageDownloaderOperation start] directly and deallocated it.
+    [self cancel];
```

Comment:
...Why not just simply use the code, but use a total `cancel` call ? This is not what it need to do.

```
 #if SD_UIKIT
         if (self.backgroundTaskId != UIBackgroundTaskInvalid) {
             // If backgroundTaskId != UIBackgroundTaskInvalid, sharedApplication is always exist
             UIApplication * app = [UIApplication performSelector:@selector(sharedApplication)];
             [app endBackgroundTask:self.backgroundTaskId];
             self.backgroundTaskId = UIBackgroundTaskInvalid;
         }
 #endif
```

Reply:
Call `cancel` in dealloc have a un-predictable effect. Because NSOperationQueue KVO your `cancelled, finished` property and have their logic. And I don't know what that `wasteful things` is.

I just mean to add just what you directly designed to solve, do not add extra code with unrelated changes. If you can sure your previous test is OK. Just remove anything in the dealloc method.